07-routing-gossip.md

```diff
@@ -462,23 +463,34 @@ of two `channel_update`s within a single second.
 
 ## Initial Sync
 
+Note that the `initial_routing_sync` feature is overridden (and should
+be considered equal to 0) by the `gossip_queries` feature if the
+latter is negotiated.
+
+Note that `gossip_queries` won't work with older nodes, so the
+value of `initial_routing_sync` is still important to control
+interactions with them.
+
 ### Requirements
 
 An endpoint node:
-  - upon establishing a connection:
-    - SHOULD set the `init` message's `initial_routing_sync` flag to 1, to
-    negotiate an initial sync.
-  - if it requires a full copy of the other endpoint's routing state:
-    - SHOULD set the `initial_routing_sync` flag to 1.
-  - upon receiving an `init` message with the `initial_routing_sync` flag set to
-  1:
-    - SHOULD send `channel_announcement`s, `channel_update`s and
-    `node_announcement`s for all known channels and nodes, as if they were just
-    received.
-  - if the `initial_routing_sync` flag is set to 0, OR if the initial sync was
-  completed:
-    - SHOULD resume normal operation, as specified in the following
-    [Rebroadcasting](#rebroadcasting) section.
+  - if the `gossip_queries` feature is negotiated:
+	- MUST NOT relay any gossip messages unless explicitly requested.
+  - otherwise:
+    - upon establishing a connection:
+      - SHOULD set the `init` message's `initial_routing_sync` flag to 1, to
```

Comment:
I'm confused by this. On this line it says to always set the flag upon establishing a connection, and on the next line it says to only set the flag if you need a full copy of the routing state.

Reply:
Yes, that was actually in the prior text.  Will fix.